lib/asciidoctor/chart/chart_block.rb

```diff
@@ -25,5 +21,28 @@ def initialize(parent, data, attrs)
         assign_caption(caption, 'figure')
       end
     end
+
+    module ChartBlockTracker
+      attr_reader :x_chart_engines
+
+      def self.extended instance
+        return if instance.instance_variable_defined? :@x_chart_engines
+
+        instance.instance_variable_set :@x_chart_engines, Set.new
+      end
```

Comment:
I extend the `Document` and then I call `get_engine` on it to resolve the engine and save  the value.
I can then retrieve the complete list of all engines used in the document.

Did I get it right?

Reply:
Yep, this looks right. My only suggestion here is to put all state in the `x_chart` instance variable to encapsulate the variables needed for this extension. I'll do the same in reducer.

Reply:
Oh right, that's a good idea 👍🏻 
I didn't understand the first time you've explained it.